pkg/kubelet/status/state/state.go

```diff
@@ -47,7 +47,7 @@ type Reader interface {
 
 type writer interface {
 	SetContainerResourceAllocation(podUID string, containerName string, alloc v1.ResourceRequirements) error
-	SetPodResourceAllocation(PodResourceAllocation) error
+	SetPodResourceAllocation(podUID string, alloc map[string]v1.ResourceRequirements) error
```

Comment:
I'd prefer to remove this method in this PR. You can define the new method in https://github.com/kubernetes/kubernetes/pull/129477#discussion_r1904749494

Reply:
I didn't mean to revert this change, I meant to delete the `SetPodResourceAllocation` method. I think with your changes in this PR this method is never called anymore, right?

Reply:
my mistake, adjusted!